spec/telegraf/rack_spec.rb

```diff
@@ -57,6 +57,16 @@
     end
   end
 
+  context 'with X-Request-Start integer timestamp in microseconds' do
+    it 'includes queue_ms value' do
+      mock.request('GET', '/', {'HTTP_X_REQUEST_START' => "t=#{Time.now.utc.to_f.to_s.delete('.')}"})
+
+      expect(last_points.size).to eq 1
+      expect(last_point.values.keys).to include 'queue_ms'
+      expect(last_point.values['queue_ms']).to match(/\A\d+\.\d+\z/)
+    end
+  end
```

Comment:
I'd definitely like to see some more specs testing with static values. So that we see that the conversion and calculation is correct for a specific seconds, milliseconds, and nanoseconds value.

Would you like to add e.g. `rubocop` to freeze the time to a known value, add some nested specs with specific values and check that the query time is as expected?

Otherwise, I will do that later.

Reply:
Oops, sorry, but I'm not a "real" Ruby dev - so I would be grateful if you fix it however you see fit.
Thank you!